war/src/main/java/io/lumeer/engine/controller/DocumentFacade.java

```diff
@@ -324,10 +284,77 @@ public void dropAttribute(final String collectionName, final String documentId,
     */
    public Set<String> getDocumentAttributes(final String collectionName, final String documentId) throws DbException {
       DataDocument dataDocument = dataStorage.readDocument(collectionName, dataStorageDialect.documentIdFilter(documentId));
-      Set<String> documentAttributes = new HashSet<>();
-      // filter out metadata attributes
-      documentAttributes.addAll(dataDocument.keySet().stream().filter(key -> !key.startsWith(LumeerConst.Document.METADATA_PREFIX)).collect(Collectors.toList()));
-      return documentAttributes;
+      return dataDocument != null ? getDocumentAttributes(dataDocument) : null;
+   }
+
+   private Set<String> getDocumentAttributes(DataDocument dataDocument) {
```

Comment:
I do not think it is necessary to add JavaDoc for private methods. You usually want to break up your code to many private methods where every method does only a single thing and its purpose is obvious from its name. In that case the JavaDoc just makes the code less readable since you need to go through a lot of comments.